conans/client/settings_preprocessor.py

```diff
@@ -0,0 +1,18 @@
+from conans.errors import InvalidConanSettingField
+
+
+def preprocess(settings, output):
+    fill_runtime(settings, output)
+
+
+def fill_runtime(settings, output):
+    try:
+        if settings.compiler == "Visual Studio":
+            if settings.get_safe("compiler.runtime") is None:
+                settings.compiler.runtime = "MDd" if settings.get_safe("build_type") == "Debug" \
+                                                  else "MD"
+                output.info("Setting 'compiler.runtime' not declared, automatically "
+                            "adjusted to '%s'" % settings.compiler.runtime)
+    except InvalidConanSettingField:  # If the settings structure doesn't match these general
```

Comment:
I guess it is somewhat related to your question about opt-it, opt-out. We need to make sure this doesn't block users, yet it is useful.

I think this might belong to the profile. So a profile can have ``runtime = auto`` or ``runtime = MT auto`` or something like that.